raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -57,15 +57,16 @@ default void handleClaim(int epoch) {}
         /**
          * Invoked after a leader has stepped down. This callback may or may not
          * fire before the next leader has been elected.
+         *
+         * @param epoch the epoch that the leader is resigning from
          */
-        default void handleResign() {}
+        default void handleResign(int epoch) {}
```

Comment:
@cmccabe Should this be a `long epoch`?  That's what `MetaLogListener.handleRenounce(long epoch) accepts.

Reply:
We would like to make it a long eventually.  In the Raft layer it's currently an int, so this is correct.